### PR TITLE
8321550: Update several runtime/cds tests to use vm flags or mark as flagless

### PR DESCRIPTION
**Pull request by calvinccheung**
Simple update to a few CDS tests:

- add `@requires vm.flagless` if the test is using `ProcessTools.createLimitedTestJavaProcessBuilder()`;
- some tests are using `ProcessTools.createLimitedTestJavaProcessBuilder()` by mistake; change it back to `ProcessTools.createTestJavaProcessBuilder()`.

Passed tiers 1 - 3 testing.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8321550](https://bugs.openjdk.org/browse/JDK-8321550): Update several runtime/cds tests to use vm flags or mark as flagless (**Enhancement** - P4)

### Reviewers
 * [Matias Saavedra Silva](https://openjdk.org/census#matsaave) (@matias9927 - Committer)
 * [Ioi Lam](https://openjdk.org/census#iklam) (@iklam - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18520/head:pull/18520` \
`$ git checkout pull/18520`

Update a local copy of the PR: \
`$ git checkout pull/18520` \
`$ git pull https://git.openjdk.org/jdk.git pull/18520/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18520`

View PR using the GUI difftool: \
`$ git pr show -t 18520`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18520.diff">https://git.openjdk.org/jdk/pull/18520.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18520#issuecomment-2023368414)

### CODE REVIEW

File: test/hotspot/jtreg/runtime/cds/TestCDSVMCrash.java [hunk 1]
```diff
@@ -49,7 +49,7 @@ public static void main(String[] args) throws Exception {
             }
         }
         // else this is the main test
-        ProcessBuilder pb = ProcessTools.createLimitedTestJavaProcessBuilder("-XX:+CrashOnOutOfMemoryError",
+        ProcessBuilder pb = ProcessTools.createTestJavaProcessBuilder("-XX:+CrashOnOutOfMemoryError",
```
Comment (matias9927) [inline test/hotspot/jtreg/runtime/cds/TestCDSVMCrash.java:52]: The arguments are no longer aligned after your change

File: test/hotspot/jtreg/runtime/cds/appcds/sharedStrings/ResolvedReferencesNotNullTest.java [hunk 2]
```diff
@@ -42,7 +42,7 @@ public static void main(String[] args) throws Exception {
         String appJar = TestCommon.getTestJar(SharedStringsUtils.TEST_JAR_NAME_FULL);
         String whiteboxParam = SharedStringsUtils.getWbParam();
 
-        ProcessBuilder pb = ProcessTools.createLimitedTestJavaProcessBuilder("-cp",
+        ProcessBuilder pb = ProcessTools.createTestJavaProcessBuilder("-cp",
```
Comment (matias9927) [inline test/hotspot/jtreg/runtime/cds/appcds/sharedStrings/ResolvedReferencesNotNullTest.java:45]: Same here

**Review by matias9927 — COMMENTED** [review]
Change looks good but I have two format concerns. When you changed `createLimitedTestJavaProcessBuilder` to `createTestJavaProcessBuilder`, the arguments are no longer aligned.

File: test/hotspot/jtreg/runtime/cds/TestCDSVMCrash.java (hunk 1 above)
Comment (calvinccheung) [reply test/hotspot/jtreg/runtime/cds/TestCDSVMCrash.java:52]: Fixed.

File: test/hotspot/jtreg/runtime/cds/appcds/sharedStrings/ResolvedReferencesNotNullTest.java (hunk 2 above)
Comment (calvinccheung) [reply test/hotspot/jtreg/runtime/cds/appcds/sharedStrings/ResolvedReferencesNotNullTest.java:45]: Fixed.

**Review by matias9927 — APPROVED** [review]
Thanks for the fix, approved!

**Review by iklam — APPROVED** [review]
LGTM

**Comment by calvinccheung** [comment]
Thanks @matias9927, @iklam for the review.

/integrate

**Pull request closed without merging**